common/lib/xmodule/xmodule/x_module.py

```diff
@@ -603,6 +605,14 @@ def get_child_by(self, selector):
                 return child
         return None
 
+    @pluggable_override('GET_UNIT_ICON_IMPL')
+    def get_icon(self):
+        """
+        An API for returning a css class identifying this module in the context of an icon.
+        It can be overridden by setting `GET_UNIT_ICON_IMPL` to an alternative implementation.
+        """
+        return self.get_icon_class()
```

Comment:
If this is defined on `XModuleMixin`, it will only apply to `XModule` instances, not `XBlock` ones, which is the newer class. Since this is where `get_icon_class` is defined too, that's probably fine, but I don't think this needs to be an `XModuleMixin` method at all... I think it could just be a top-level method in some runtime/xblock utils file, e.g.

```python
@pluggable_override('GET_UNIT_ICON_IMPL')
def get_icon_for_block(block):
    return block.get_icon_class()
```

No need to change now, I'm not an official reviewer here, just pointing out options.

Reply:
I was wondering about the correct place for this, but wasn't sure how this could be applied to `XBlock`, as it doesn't have a method for getting an icon. Should we just raise `NotImplementedError` when we don't find `block.get_icon_class` method?

Reply:
I think it should be a top level function in [`xblock_utils/__init__.py`](https://github.com/edx/edx-platform/blob/master/openedx/core/lib/xblock_utils/__init__.py).

Reply:
Sounds good, I'll move this.